app/src/main/java/com/exallium/mvvmcleanexample/app/users/UserEditView.kt

```diff
@@ -25,34 +26,46 @@ class UserEditView : AppCompatActivity() {
     @Inject
     lateinit var router: AppRouter
 
-    val compositeDisposable = CompositeDisposable()
+    private var binding by autoEcoBinding<UserEditViewBinding>(null)
+    private var disposable by ecoDisposable(null)
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        val binding = DataBindingUtil.setContentView<UserEditViewBinding>(this, R.layout.user_edit_view)
+        binding = DataBindingUtil.setContentView<UserEditViewBinding>(this, R.layout.user_edit_view).also { binding ->
```

Comment:
is there any way we can set the binding then do the rest of setup outside? I'm just not a huge fan of extra levels of scoping.